pkg/lspext/cacheext.go

```diff
@@ -0,0 +1,23 @@
+package lspext
+
+// CacheGetParams is the input for 'cache/xget'. The response is a
+// 'CacheItem'. This cache is global to a language server. IE it is shared
+// amongst workspaces, but not amongst different language server types.
+type CacheGetParams struct {
+	Key string `json:"key"`
+}
+
+// CacheItem is the response for 'cache/xget'.
+type CacheItem struct {
+	// Ok is true if the item was successfuly retrieved from the cache.
+	Ok bool `json:"ok"`
+	// Value is the value stored in the cache. It is present if Ok is true.
+	Value string `json:"value,omitempty"`
+}
+
+// CacheSetParams is the input for 'cache/xset'. It is a notify method, so
+// does not have a response.
+type CacheSetParams struct {
+	Key   string `json:"key"`
+	Value string `json:"value"`
```

Comment:
I see your point, but it would require the JS/TS LS for example to `JSON.stringify()` the object on the server side, basically double-encoding it. This also means that you will have to double-escape everything, so it will result in more bytes being sent over the wire.

For my own use case, it makes no difference. I will use PHP's own serialization to keep class information.